avm/res/recovery-services/vault/main.bicep

```diff
@@ -0,0 +1,458 @@
+metadata name = 'Recovery Services Vaults'
+metadata description = 'This module deploys a Recovery Services Vault.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Required. Name of the Azure Recovery Service Vault.')
+param name string
+
+@description('Optional. The storage configuration for the Azure Recovery Service Vault.')
+param backupStorageConfig object = {}
+
+@description('Optional. Enable/Disable usage telemetry for module.')
+param enableTelemetry bool = true
+
+@description('Optional. Location for all resources.')
+param location string = resourceGroup().location
+
+@description('Optional. List of all backup policies.')
+param backupPolicies array = []
+
+@description('Optional. The backup configuration.')
+param backupConfig object = {}
+
+@description('Optional. List of all protection containers.')
+@minLength(0)
```

Comment:
is the @minLength(0) necessary? I think it's a default? This also applies to other array parameters.